netbox/extras/management/commands/reindex.py

```diff
@@ -81,12 +81,19 @@ def handle(self, *model_labels, **kwargs):
             self.stdout.write(f'  {app_label}.{model_name}... ', ending='')
             self.stdout.flush()
 
-            if kwargs['lazy']:
-                content_type = ContentType.objects.get_for_model(model)
-                if cached_count := search_backend.count(object_types=[content_type]):
+            content_type = ContentType.objects.get_for_model(model)
+            cached_count = search_backend.count(object_types=[content_type])
+
+            if cached_count:
+                if kwargs['lazy']:
                     self.stdout.write(f'Skipping (found {cached_count} existing).')
                     continue
 
+                if model_labels:
+                    self.stdout.write(f'clearing {cached_count} cached values... ', ending='')
+                    self.stdout.flush()
+                    search_backend.clear(object_types=[content_type])
```

Comment:
We want to keep the cache clearing logic before and separate from the indexing logic. The call to `search_backend.clear()` above (originally line 73) should be wrapped in a conditional that checks for `model_labels` and passes either the appropriate content types or None. The indexing logic can then be retained in its original form.